helper/resource/testing.go

```diff
@@ -855,6 +861,21 @@ type ConfigPlanChecks struct {
 	PostApplyPostRefresh []plancheck.PlanCheck
 }
 
+// ConfigQueryChecks defines the different points in a Config TestStep when query checks can be run.
+type ConfigQueryChecks struct {
```

Comment:
nit: I don't think this is used anywhere? It seems to suggest that the query command would be run at different stages of the `Config` mode, but `Query` looks like it's own "test mode"